Orbit/UI/ViewModels/MessagingViewModel.swift

```diff
@@ -0,0 +1,209 @@
+//
+//  MessagingViewModel.swift
+//  Orbit
+//
+//  Created by Jessy Tseng on 2024-11-02.
+//
+
+@preconcurrency import Appwrite
+import Foundation
+import Appwrite
+
+class MessagingViewModel: ObservableObject{
+    private var messagingService: MessagingServiceProtocol = MessagingService()
+    private var userManagementService: UserManagementServiceProtocol = UserManagementService()
+    private var subscription: RealtimeSubscription?
+    
+    @MainActor
+    func getConversations(_ accountId: String) async -> [String]{
+        do{
+            if let userModel = try await userManagementService.getUser(accountId){
+                if let conversations = userModel.data.conversations{
+                    return conversations
+                }
+            }
+            throw NSError(domain: "UsersNotFound: \"\(accountId)\"", code: 404)
+        } catch {
+            print("MessagingViewModel - getConversations failed \(error.localizedDescription)")
+            return []
+        }
+    }
+    
+    
+    @MainActor
+    func createConversation(_ participants: [String]) async {
+        do{
+            // Create entry in conversation table
+            let conversationData = ConversationModel(participants: participants)
+            let conversationEntry = try await messagingService.createConversation(conversationData)
+            
+            // Add conversation id to users
+            for accountId in participants{
+                if let userModel  = try await userManagementService.getUser(accountId) {
+                    if let conversations = userModel.data.conversations{
+                        var newConversations = conversations
+                        newConversations.append(conversationEntry.id)
+                        let newUserModel = UserModel(
+                            accountId: accountId,
+                            name: userModel.data.name,
+                            interests: userModel.data.interests,
+                            latitude: userModel.data.latitude,
+                            longitude: userModel.data.longitude,
+                            isInterestedToMeet: userModel.data.isInterestedToMeet,
+                            conversations: newConversations
+                        )
+                        try await userManagementService.updateUser(accountId: accountId, updatedUser: newUserModel)
```

Comment:
updateUser doesn't have to have to have the whole UserModel. We can just pass the values we wanna change. Also do we really need to store the list of conversations in the user collection? Can't we just get the conversationId associated with a user by getting the `participants` from the conversations collection?

Reply:
I tried that approach with updateUser at first but the model did not allow me to modify a single field. Storing the list of conversations in user is just easier to query vs query from conversation collection. If you really want it in conversation collection instead I can look into it.